### DISCUSSION TITLE
20220719

### DISCUSSION THREAD

**Discussion by tmiw** [ARSFI/paclink-open-source#31] [General]
This is a pre-production build intended to gauge current status of the project and determine what needs to be fixed. Brief testing indicates that packet (using Kenwood TM-D700A), AGW (using AGWPE and UZ7HO soundmodem) and Telnet channels work; Pactor is TBD.

Contains (vs. [last pre-release version](https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20220703)):

* Pactor: apply test fix to see if Pactor connect dialog reopens more reliably.

Instructions:

1. Unzip paclink-open-source.zip somewhere on your hard drive.
2. Run setup.exe to install on your system. Paclink will be installed in the Start menu.
3. Open Paclink from the Start menu and reconfigure if required. (There is no import functionality from the existing Paclink at this time but if you used a pre-release version, configuration should be saved.)

Please leave comments in the release's discussion thread or email K6AQ@winlink.org with feedback. 👍
 

<hr /><em>This discussion was created from the release <a href='https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20220719'>20220719</a>.</em>

**Comment by kd4vrf** [comment]
While testing this release I haven't seen any of the Pactor dialog box issues. It's been running for almost two weeks and I haven't seen the dialog box fail to open even once in that time. It looks like that one is resolved.

I tried a couple of different email clients to troubleshoot some stripping of line returns in emails from Hamword. I was using Microsoft Mail and there aren't many settings for the reading pane, so I tried Thunderbird and Mailbird just to see if it was the client stripping the line returns away. I haven't gotten to the bottom of the line return stripping yet but in the process of experimenting with the other clients setup I changed my POP3 password and now I'm not able to authenticate.

After I changed the password I was able to use the new password on both clients, I was even able to use the new password to telnet to the POP3 server in Paclink. However, after I shut down my Windows VM and restarted it this morning I'm no longer able to authenticate to the Paclink POP3 server.

I have tried the following to resolve:

- Changing the POP3 password again
- Removed the callsign account and recreated it in Paclink
- Uninstalled Paclink and reinstalled
- Uninstalled Packlink, deleted "packing.db", and reinstalled

None of the above steps resulting in me being able to login to the POP3 server.

Here is a screen shot of the fresh install of Paclink with a fresh "packing.db" and a newly created account failing. You can see what I set the temporary password to in the "Manage Callsign Accounts" dialog box in the screenshot and the telnet session.

![Telnet to POP3 Server - Screen Shot 2022-08-02 at 11 24 07 AM](https://user-images.githubusercontent.com/45251519/182412138-7e14b3eb-6a65-4b8d-b539-0504c1641158.png)

I've also noticed that in this build when I click "settings --> edit account profile" the dialog box doesn't open so I'm not able to view or edit any of those settings after the initial setup. I'm not sure if that is new behavior, I haven't tried to modify any of those settings Since I initially setup Paclink months ago.

Here are the Paclink logs: 
[Paclink Logs 8-2-2022.zip](https://github.com/ARSFI/paclink-open-source/files/9243843/Paclink.Logs.8-2-2022.zip)

Let me know if you need anything else.

**Comment by k0qed** [comment]
The POP3 session is converting the entered password to upper case before
attempting to validate it. This is the method that used to be used before
Winlink switched to case sensitive passwords. The bug was probably
introduced in this project with the changes for saving settings to a db
rather than the ini file.

I suggest that we use case sensitive passwords for the pop3 accounts (and
others). For the pop3 session it appears to be a one line change --
omitting the call to ToUpper() on line 180.

-Lee

**Reply by tmiw** [reply]
Should the username still be capitalized, or should ToUpper for that be removed too?

**Comment by k0qed** [comment]
There's a similar problem with the SMTP session ~ line 318 has the
conversion to upercase.

-Lee

**Comment by kd4vrf** [comment]
If I create a callsign account using all upper case then it will allow me to authenticate and retrieve messages via telnet.

See below:
![Upper Case Pass - Screen Shot 2022-08-02 at 2 59 41 PM](https://user-images.githubusercontent.com/45251519/182453604-660c6856-392e-4cbb-85e6-8940be1c9c25.png)

It certainly seems like that conversion to upper case is the culprit. My original password was upper case, not sure if that was dumb luck, the GUI used to force it as upper case, or it was something I did at the time of my first install to troubleshoot. All upper case passwords are not the norm for me when creating passwords.

**Comment by k0qed** [comment]
I think the comparison of username should be case insensitive. If I
remember correctly, usernames are callsigns or tactical addresses. If
that's the case, then uppercase is probably the best way to store them.

-Lee

**Comment by tmiw** [comment]
New release with the changes proposed by @k0qed: https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20220802.
